src/amazon/dsstne/engine/NNNetwork.cpp

```diff
@@ -2319,7 +2319,7 @@ bool NNNetwork::Validate()
     const NNFloat alpha  = (NNFloat)1.0;
     const NNFloat lambda = (NNFloat)0.0; // regularization parameter (no need for bias test)
     const NNFloat mu     = (NNFloat)0.0; // no momentum
-    const NNFloat epsilon = delta * 10.f;
+    const NNFloat epsilon = delta * 20.f;
```

Comment:
There are couple of issues with numerical gradient validation:
- The deeper network the higher numerical error in the cost function evaluation;
- Current implementation uses non symmetrical derivative and it is not the best choice (in the code I put TODO).
Because of these two reasons threshold tuning becomes empirical, and the goal is choose value as small as possible so that probability of unit test failure in case when gradient implementation is correct should be zero and probability of detection of incorrect gradient implementation should be as high as possible. I started testing deeper networks and with threshold 10 there were failures even though gradient implementation is correct. So I set it back to 20